jenkins/shared-library/vars/createMultibranchJob.groovy

```diff
@@ -43,6 +44,39 @@ void call(Map jobConfiguration) {
 										contextLabel(notificationContextLabel)
 										typeSuffix(false)
 									}
+
+									cleanBeforeCheckoutTrait {
+										extension {
+											// Deletes untracked submodules and any other subdirectories which contain .git directories.
+											deleteUntrackedNestedRepositories(true)
+										}
+									}
+
+									submoduleOptionTrait {
```

Comment:
why is this needed ?
is this cloning the submodules ?

Reply:
Yea, Jenkins needs access to the Dockerfiles in ``jenkins/docker`` to build the image before the stages in pipeline. I needed to clone the submodules.

Reply:
is there a way to only clone a subset of the repo using sparsecheckout, like init.sh is doing?

Reply:
I was looking last night but havent found a way yet.  Will continue looking today.  Hopefully with fresh eyes might have better luck :)

Reply:
I havent found a good way to checkout the submodule using sparsecheckout.  The git plug doesnt support it for submodules(https://issues.jenkins.io/browse/JENKINS-42628)
Workaround would be
1. Use git hooks(post-checkout) which runs the init.sh
2. In the pipeline, create a new stage that will run init.sh script before the Dockerfile.  This will move the Dockerfile into the stages
3. Could probably enable ``Shallow clone`` to save some time and space.
@Jaguar0625 @gimre-xymcity any thoughts?

Reply:
i think this is ok for now, but you can try shallow clone and see if it improves things. if so, you can make another PR with those changes. 👍